Remove custom 'createOrAlter' field in ViewStmt and grammar

R4hul04 opened the pull request:
This PR modifies how we identify T-SQL `ALTER VIEW` and `CREATE OR ALTER VIEW` operations. Instead of using a custom field added to the PostgreSQL ViewStmt structure, we now use the standard options list mechanism instead

This PR includes following changes:

- Updated grammar rules to add a marker in the options list
- Added logic to check for this option in alter view operation handling

Engine PR - https://github.com/babelfish-for-postgresql/postgresql_modified_for_babelfish/pull/607
### Issues Resolved

BABEL-5939

Signed off by: Rahul Parande rparande@amazon.com

### Test Scenarios Covered ###
* **Use case based -**


* **Boundary conditions -**


* **Arbitrary inputs -**


* **Negative test cases -**


* **Minor version upgrade tests -**


* **Major version upgrade tests -**


* **Performance tests -**


* **Tooling impact -**


* **Client tests -**



### Check List
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

Commit:
Removed createOrAlter' field in ViewStmt and grammar files and instead
use Viewstmt->options

Signed-off-by: Rahul Parande <rparande@amazon.com>

Commit:
Removed call to handle_bbf_view_binding_on_object_drop during alter
function

Signed-off-by: Rahul Parande <rparande@amazon.com>

Commit:
Fixed spacing in grammar file

Signed-off-by: Rahul Parande <rparande@amazon.com>

tanscorpio7 commented on contrib/babelfishpg_tsql/src/hooks.c:
```diff
@@ -7337,15 +7337,27 @@ handle_bbf_view_binding_on_object_drop(const ObjectAddress *droppedObject, Relat
+	ListCell 		*option;
+	bool 			tsql_alter_view_op = false;
+
+	if (stmt != NULL)
+	{
+		foreach(option, stmt->options)
+		{
+			DefElem *def = (DefElem *) lfirst(option);
+			
+			if (strcmp(def->defname, "tsql_alter_view_op") == 0)
+			{
+				tsql_alter_view_op = true;
+				break;
+			}
+		}
+	}
```
Can we instead convert `ViewStmt *stmt` in function signature to a boolean `is_alter_view`
Also is `Relation depRel` being used by anyone ?

R4hul04 replied:
Thanks for the observation.  depRel  isn't used anywhere, I might have missed to remove it. Also changed the argument to bool instead of Viewstmt

Commit:
Changed handle_bbf_view_binding_on_object_drop() input argument to take
boolean instead of ViewStmt

Signed-off-by: Rahul Parande <rparande@amazon.com>

Commit:
Fixed is_alter_view flag in if condition

Signed-off-by: Rahul Parande <rparande@amazon.com>

Commit:
Trigger Build

Commit:
Trigger Build

shardgupta committed:
Update manual_tags workflow to use intermediate variable to fix possible (#4364)

Update manual_tags workflow to use intermediate variable to fix possible script injection

Task: NO-JIRA

Signed-off-by: Shard Gupta shardga@amazon.com
